presto-main/src/main/java/io/prestosql/server/DynamicFilterService.java

```diff
@@ -83,33 +78,29 @@
 import static io.prestosql.sql.planner.SystemPartitioningHandle.SOURCE_DISTRIBUTION;
 import static io.prestosql.util.MorePredicates.isInstanceOfAny;
 import static java.util.Objects.requireNonNull;
-import static java.util.concurrent.Executors.newSingleThreadScheduledExecutor;
-import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.Executors.newSingleThreadExecutor;
 
 @ThreadSafe
 public class DynamicFilterService
 {
-    private final Duration dynamicFilteringRefreshInterval;
-    private final ScheduledExecutorService collectDynamicFiltersExecutor = newSingleThreadScheduledExecutor(daemonThreadsNamed("DynamicFilterService"));
-
+    private final ExecutorService executor;
     private final Map<QueryId, DynamicFilterContext> dynamicFilterContexts = new ConcurrentHashMap<>();
 
-    @Inject
-    public DynamicFilterService(FeaturesConfig featuresConfig)
+    public DynamicFilterService()
     {
-        this.dynamicFilteringRefreshInterval = requireNonNull(featuresConfig, "featuresConfig is null").getDynamicFilteringRefreshInterval();
+        this(newSingleThreadExecutor(daemonThreadsNamed("DynamicFilterService")));
```

Comment:
It's a good point. This is to be tested, but I think I will improve this in subsequent PR. Generally, what happens here should not be that CPU intensive.

Reply:
If it's not CPU intensive, does it need to be done asynchronously?

Reply:
I think it needs more threads now, since there is potentially expensive `union` operation. Plus we might want to de-cast DFs which would involve calling `satured_floor_casts`